slab/stair infestation and block datagen

PiggyDr opened the pull request:
adds infested variants for stairs, slabs, and walls, and tag infested variants for fences and fence gates. also there is datagen if you want to use it but it doesn't iterate over the block registry or anything so you can just ignore it

PiggyDr committed:
get gradle to work

i dont know how gradle works and can't be bothered to learn so for now i'll just do this and revert it before merge

PiggyDr committed:
infested wood stairs loot table

PiggyDr committed:
loot tables for all infested stairs

PiggyDr committed:
dev conversion wand now actually works

PiggyDr committed:
conversion wand can cure blocks

PiggyDr committed:
stair variants for all existing stairs made of an explicitly infestable material

PiggyDr committed:
default normal variant for tag infestation entries

PiggyDr committed:
add new stairs + end stone to creative tab

-renamed infested endstone
-added item models for stairs
-added stair lang entries

PiggyDr committed:
tag stairs properly

PiggyDr committed:
sturdy stairs + missing stair loot tables

PiggyDr committed:
infested crumbling stairs + fix stair models

PiggyDr committed:
explicit slabs, infested mud brick blocks, no floating grass

PiggyDr committed:
tag infested slabs

PiggyDr committed:
infested stone/blackstone brick blocks + tried datagen, works

PiggyDr committed:
clean up datagen

PiggyDr committed:
infested blackstone texture

PiggyDr committed:
infested walls + more datagen

PiggyDr committed:
remove blackstone walls from datagen

PiggyDr committed:
tag infested walls

PiggyDr committed:
fix typo

PiggyDr committed:
infested fences

PiggyDr committed:
infested fence gates

PiggyDr committed:
fixes

PiggyDr committed:
it is time

PiggyDr committed:
Merge branch '1.20' into 1.20-block-infestation

Mikeatron-User commented:
Wow, thanks for the contribution, I will review this to make sure everything works correctly :)

PiggyDr committed:
some comments (should be last commit)

Mikeatron-User commented:
I will merge this once I'm finalizing the next major update.

Mikeatron-User committed:
Added Missing Block Infection Entry for Wooden Stairs

Mikeatron-User committed:
Bamboo Now Accounted For BlockInfestation

Modified Wooden Slab entry to instead use Slab block tag instead of wooden slab tag. Doing this allows for bamboo to be accounted for.

Mikeatron-User commented:
Fantastic Work! I made a fix for wooden stairs and an edit to account for the bamboo set of blocks. Other than that, great work!
